### PR TITLE
chore: P4PU-188 added exception handler class

### PR DESCRIPTION
**Pull request by Giuseppe-LaManna**
#### List of Changes
- added class exception handler
- added swagger 
- added openapi generator dependency

#### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: src/main/java/it/gov/pagopa/arc/exception/ArcExceptionHandler.java [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+    }
+    @ExceptionHandler(TooManyRequestException.class)
+    public ResponseEntity<ArcErrorDTO> handleTooManyRequestException(RuntimeException ex, HttpServletRequest request){
+        return handleArcErrorException(ex, request, HttpStatus.TOO_MANY_REQUESTS, ArcErrorDTO.ErrorEnum.TOO_MANY_REQUEST);
+    }
+    @ExceptionHandler(InternalServerException.class)
+    public ResponseEntity<ArcErrorDTO> handleInternalServerException(RuntimeException ex, HttpServletRequest request){
+        return handleArcErrorException(ex, request, HttpStatus.INTERNAL_SERVER_ERROR, ArcErrorDTO.ErrorEnum.INTERNAL_SERVER_ERROR);
+    }
+    private static ResponseEntity<ArcErrorDTO> handleArcErrorException(RuntimeException ex, HttpServletRequest request, HttpStatus httpStatus, ArcErrorDTO.ErrorEnum errorEnum) {
```
Comment (and-mora) [inline src/main/java/it/gov/pagopa/arc/exception/ArcExceptionHandler.java:59]: is the reference to 'Arc' necessary?

File: src/main/java/it/gov/pagopa/arc/exception/custom/InternalServerException.java
Comment (and-mora) [inline src/main/java/it/gov/pagopa/arc/exception/custom/InternalServerException.java:1]: business exceptions should **never** codify a http status code.

File: src/main/java/it/gov/pagopa/arc/exception/custom/TooManyRequestException.java
Comment (and-mora) [inline src/main/java/it/gov/pagopa/arc/exception/custom/TooManyRequestException.java:1]: business exceptions should **never** codify a http status code.

File: src/main/java/it/gov/pagopa/arc/exception/custom/UnauthorizedException.java
Comment (and-mora) [inline src/main/java/it/gov/pagopa/arc/exception/custom/UnauthorizedException.java:1]: business exceptions should **never** codify a http status code.

**Review by and-mora — CHANGES_REQUESTED** [review]

File: src/main/java/it/gov/pagopa/arc/exception/ArcExceptionHandler.java (hunk 1 above)
Comment (Giuseppe-LaManna) [reply src/main/java/it/gov/pagopa/arc/exception/ArcExceptionHandler.java:59]: it's just a name, it's not necessary, I can remove the "Arc" reference from the class name

File: openapi/pagopa-arc-be.openapi.yaml [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+      tags:
+        - arc transactions
+      summary: "Retrieve the paged transaction list from arc"
+      operationId: getTransactionList
+      parameters:
+        - name: x-fiscal-code
+          in: header
+          required: true
+          schema:
+            type: string
```
Comment (and-mora) [inline openapi/pagopa-arc-be.openapi.yaml:21]: is this parameter not inferred from the session cache?

**Review by and-mora — CHANGES_REQUESTED** [review]

File: openapi/pagopa-arc-be.openapi.yaml (hunk 2 above)
Comment (Giuseppe-LaManna) [reply openapi/pagopa-arc-be.openapi.yaml:21]: yes it is. 
for the moment that we don't yet have this implementation we pass the cf directly from BE, correct?

**Review by and-mora — APPROVED** [review]

**Pull request merged**